compiler/src/dotty/tools/dotc/core/MatchTypeTrace.scala

```diff
@@ -91,6 +98,9 @@ object MatchTypeTrace:
          |  matches none of the cases
          |
          |    ${casesText(cases)}"""
+    case EmptyScrutinee(scrut) =>
+      i"""  failed since selector  $scrut
+         |  is uninhabited."""
```

Comment:
Could we explain a bit what uninhabited means? Something like `uninhabited (no instance of that type can be constructed)`